subgraph/src/public-lock.ts

```diff
@@ -440,11 +441,17 @@ export function createReceipt(event: ethereum.Event): void {
       // including one for the ERC20 transfer
       for (let i = 0; i < logs.length; i++) {
         const txLog = logs[i]
-        if (
-          txLog.address == tokenAddress &&
-          // Do we always have txLog.topics[0] ?
-          txLog.topics[0].toHexString() == ERC20_TRANSFER_TOPIC0
-        ) {
+
+        const hashedSignature = txLog.topics[0].toHexString()
+        const decodedSignature = ethereum.decode(
+          '(address,address,uint256)',
+          txLog.data
+        )
```

Comment:
```suggestion
        const decodedSignature = ethereum.decode(
          '(address,address,uint256)',
          txLog.data
        )
```

@julien51 @clemsos can i review here?